File: src/main/java/ai/devchat/devchat/handler/GetKeyRequestHandler.java
package ai.devchat.devchat.handler;

import ai.devchat.devchat.ActionHandler;
import ai.devchat.devchat.DevChatActionHandler;
import ai.devchat.idea.storage.SensitiveDataStorage;
import com.alibaba.fastjson.JSONObject;

public class GetKeyRequestHandler implements ActionHandler {
    private JSONObject metadata;
    private JSONObject payload;
    private final DevChatActionHandler devChatActionHandler;

    public GetKeyRequestHandler(DevChatActionHandler devChatActionHandler) {
        this.devChatActionHandler = devChatActionHandler;
    }

    @Override
    public void executeAction() {
        String callbackFunc = metadata.getString("callback");
        String key = SensitiveDataStorage.getKey();

        if (key != null && !key.isEmpty()) {
            devChatActionHandler.sendResponse("getKey/response", callbackFunc, (metadata, payload) -> {
                metadata.put("status", "success");
                metadata.put("error", "");
                payload.put("key", key);
            });
        } else {
            devChatActionHandler.sendResponse("getKey/response", callbackFunc, (metadata, payload) -> {
                metadata.put("status", "error");
                metadata.put("error", "key is empty");
            });
        }
    }

    @Override
    public void setMetadata(JSONObject metadata) {
        this.metadata = metadata;
    }

    @Override
    public void setPayload(JSONObject payload) {
        this.payload = payload;
    }
}

File: src/main/kotlin/ai/devchat/devchat/handler/GetKeyRequestHandler.kt
package ai.devchat.devchat.handler

import ai.devchat.devchat.BaseActionHandler
import ai.devchat.devchat.DevChatActions
import ai.devchat.idea.storage.SensitiveDataStorage
import com.alibaba.fastjson.JSONObject

class GetKeyRequestHandler(requestAction: String, metadata: JSONObject?, payload: JSONObject?) : BaseActionHandler(
    requestAction,
    metadata,
    payload
) {
    override val actionName: String = DevChatActions.GET_KEY_RESPONSE
    override fun executeAction() {
        val key = SensitiveDataStorage.key
        if (key.isNullOrEmpty()) {
            throw RuntimeException("key is empty")
        }
        send(payload = mapOf("key" to key))
    }
}
